runtime/android/core_internal/src/org/xwalk/core/internal/XWalkMediaPlayerResourceLoadingFilter.java

```diff
@@ -39,10 +45,20 @@ public boolean shouldOverrideResourceLoading(MediaPlayer mediaPlayer,
                     context.getAssets().openFd(AndroidProtocolHandler.getAssetPath(uri));
             mediaPlayer.setDataSource(
                     afd.getFileDescriptor(), afd.getStartOffset(), afd.getLength());
-
             return true;
         } catch (Exception e) {
             return false;
         }
     }
+
+    @Override
+    public ExternalMediaPlayer getExternalMediaPlayer() {
+        ExternalMediaPlayer exMediaPlayer = mContentsClientBridge.getExternalMediaPlayer();
+
+        if (exMediaPlayer == null) {
+            exMediaPlayer = new ExternalMediaPlayer();
```

Comment:
- If the media player in `XWalkContentsClientBridge` is null, everytime you call `getExternalMediaPlayer` from `XWalkMediaPlayerResourceLoadingFilter`, a new media player will be created. Is that alright?

- After created new media player here, the result of `XWalkMediaPlayerResourceLoadingFilter.getExternalMediaPlayer` is different from the result of `XWalkContentsClientBridge.getExternalMediaPlayer`, a little confused for me.

Maybe you should implement this in `XWalkContentsClientBridge.getExternalMediaPlayer`?